manifests/init.pp

```diff
@@ -1,151 +1,59 @@
+  Boolean $fqdn_entry = true,
+  Stdlib::IP::Address $fqdn_ip = $::ipaddress,
+  Variant[String, Array[String, 1]] $fqdn_host_aliases = $::hostname,
+  Boolean $purge_hosts = false,
+  Optional[Hash] $host_entries = undef,
+  Optional[Stdlib::Absolutepath] $target = undef,
 ) {
 
-
-  # validate type and convert string to boolean if necessary
-  if is_string($collect_all) {
-    $collect_all_real = str2bool($collect_all)
-  } else {
-    $collect_all_real = $collect_all
-  }
-
-  # validate type and convert string to boolean if necessary
-  if is_string($enable_ipv4_localhost) {
-    $ipv4_localhost_enabled = str2bool($enable_ipv4_localhost)
-  } else {
-    $ipv4_localhost_enabled = $enable_ipv4_localhost
-  }
-
-  # validate type and convert string to boolean if necessary
-  if is_string($enable_ipv6_localhost) {
-    $ipv6_localhost_enabled = str2bool($enable_ipv6_localhost)
-  } else {
-    $ipv6_localhost_enabled = $enable_ipv6_localhost
-  }
-
-  # validate type and convert string to boolean if necessary
-  if is_string($enable_fqdn_entry) {
-    $fqdn_entry_enabled = str2bool($enable_fqdn_entry)
-  } else {
-    $fqdn_entry_enabled = $enable_fqdn_entry
-  }
-
-  # validate type and convert string to boolean if necessary
-  if is_string($use_fqdn) {
-    $use_fqdn_real = str2bool($use_fqdn)
-  } else {
-    $use_fqdn_real = $use_fqdn
-  }
-
-  # validate type and convert string to boolean if necessary
-  if is_string($purge_hosts) {
-    $purge_hosts_enabled = str2bool($purge_hosts)
-  } else {
-    $purge_hosts_enabled = $purge_hosts
-  }
-
-  if $ipv4_localhost_enabled == true {
-    $localhost_ensure     = 'present'
-    $localhost_ip         = '127.0.0.1'
-    $my_localhost_aliases = $localhost_aliases
-  } else {
-    $localhost_ensure     = 'absent'
-    $localhost_ip         = '127.0.0.1'
-    $my_localhost_aliases = undef
-  }
-
-  if $ipv6_localhost_enabled == true {
-    $localhost6_ensure     = 'present'
-    $localhost6_ip         = '::1'
-    $my_localhost6_aliases = $localhost6_aliases
-  } else {
-    $localhost6_ensure     = 'absent'
-    $localhost6_ip         = '::1'
-    $my_localhost6_aliases = undef
-  }
-
-  if !is_string($my_localhost_aliases) and !is_array($my_localhost_aliases) {
-    fail('hosts::localhost_aliases must be a string or an array.')
-  }
-
-  if !is_string($my_localhost6_aliases) and !is_array($my_localhost6_aliases) {
-    fail('hosts::localhost6_aliases must be a string or an array.')
-  }
-
-  if $fqdn_entry_enabled == true {
+  if $fqdn_entry == true {
     $fqdn_ensure          = 'present'
-    $my_fqdn_host_aliases = $fqdn_host_aliases
-    $fqdn_ip              = $::ipaddress
+    $_fqdn_host_aliases = $fqdn_host_aliases
   } else {
     $fqdn_ensure          = 'absent'
-    $my_fqdn_host_aliases = []
-    $fqdn_ip              = $::ipaddress
+    $_fqdn_host_aliases = undef
   }
 
-  Host {
-    target => $target,
+  $host_defaults = {
+    'target' => $target,
   }
```

Comment:
Why was this change done? with only one default, wouldn't it be easier to just put the target parameter in each host resource?